tox.ini

```diff
@@ -3,6 +3,7 @@ envlist = flake8, py{36,37}, benchmark, benchmark-cuda{101,102}, mypy
 
 [testenv]
 commands=
+    python -m ipykernel install --user --name libertem-kernel
```

Comment:
Using `--user` here means running the tests will change the user's ipython configuration, and may conflict with an existing ipython kernel called `libertem-kernel` (it creates a directory under `~/.local/share/jupyter/kernels`). Is it maybe possible to use `ipykernel install --sys-prefix` instead? That would only modify the tox environment and put the kernel configuration into `$VIRTUAL_ENV/share/jupyter/kernels`, which would not conflict with the user's configuration.

Running the tests outside of tox (with plain pytest) seems to work for me, too, without having the jupyter kernel installed, or is that just coincidence? Also, you have `kernel='libertem-env'` in the tests, and create `libertem-kernel` here, so there seems to be some mismatch. As a note, when running via tox, I get the following in the logs:

```
2020-07-22 11:56:11 INFO Executing notebook with kernel: python
```

Reply:
> Running the tests outside of tox (with plain pytest) seems to work for me, too, without having the jupyter kernel installed, or is that just coincidence?

I already had a jupyter kernel `libertem-env`, so I was able to run plain pytest without any problem. When I tried to run the test using tox, it was failing because of the missing kernel. So I added the instruction to set up a new jupyter kernel and all tests were working fine.

Reply:
> 2020-07-22 11:56:11 INFO Executing notebook with kernel: python

I too get this log message, even after matching the kernel name. I have no idea how this happens.

Reply:
I also get the following warning:


```
tests/template/test_sd_template.py::test_sd_default
tests/template/test_sd_template.py::test_sd_roi
  /home/alex/.virtualenvs/notebook-generator/lib/python3.8/site-packages/traitlets/config/configurable.py:73: DeprecationWarning: Passing unrecoginized arguments to super(ExecutePreprocessor).__init__(kernel='libertem-env').
  object.__init__() takes exactly one argument (the instance to initialize)
  This is deprecated in traitlets 4.2.This error will be raised in a future release of traitlets.
    super(Configurable, self).__init__(**kwargs)
```

From [this example](https://nbconvert.readthedocs.io/en/latest/execute_api.html#example), the parameter should be called `kernel_name`, but then I wonder how installing the kernel in tox.ini fixed things for you. Can you maybe try to remove the kernel installation from tox.ini, remove the kernel from your jupyter config (`~/.local/share/jupyter/kernels` or similar), remove the `kernel='...'` from the `ExecutePreprocessor` instantiation and see what happens, inside of tox and outside? Maybe something unrelated changed and made things work for you.